phangsPipeline/handlerImagingChunked.py

```diff
@@ -0,0 +1,1521 @@
+            '''
+            Gather the chunked data into a final cube.
+            '''
+
+            if root_name is not None:
+                root_name_label = "_{}".format(root_name)
+            else:
+                root_name_label = ""
+
+            # These are the final cube names. We'll loop over cube products to concat together
+            fname_dict = self._fname_dict(imagename="{0}{1}".format(self.image_root, root_name_label),
+                                          imaging_method=self.imaging_method)
+
+            del fname_dict['root']
+            del fname_dict['suffix']
+
+            # Make dictionary to track existing chunked image names
+            chunk_fname_dict = dict.fromkeys(fname_dict)
+
+            # We can't concat non cube products:
+            skip_types = ['sumwt', 'weight']
+
+            # With the chunk names sorted, concat into cubes
+            for img_type in fname_dict:
+
+                chunk_fname_dict[img_type] = []
+
+                missing_chunks = []
+
+                logger.info("Concatenating chunks of type {} into a final cube".format(img_type))
+
+                for chunk_num in self.chunk_params:
+
+                    this_imagename = "{0}{1}.{2}".format(self.chunk_params[chunk_num]['full_imagename'],
+                                                        root_name_label, img_type)
+
+                    if not os.path.exists(this_imagename):
+                        missing_chunks.append(chunk_num)
+                        continue
+
+                    chunk_fname_dict[img_type].append(this_imagename)
+
+                if img_type in skip_types:
+                    continue
+
+                if len(chunk_fname_dict[img_type]) != self.nchunks:
+                    logger.error("Existing imaging products for type {} do not match the expected number of chunks.".format(img_type) +
+                                "This cube will not be made.")
+                    continue
+
+                if len(missing_chunks) > 0:
+                    logger.error("Missing the following chunks for the {0}: {1}".format(img_type, missing_chunks))
+                    logger.error("The concatenated cube cannot be made for {}".format(img_type))
+                    continue
+
+                if os.path.exists(fname_dict[img_type]):
+                    if overwrite:
+                        os.system("rm -rf {}".format(fname_dict[img_type]))
+                    else:
+                        logger.warn("Cube name {} already exists. Skipping.".format(fname_dict[img_type]) +
+                                    " Enable overwrite=True to force writing a new version.")
+                        continue
+
+                logger.info("Making {0} from this list of chunked images:".format(fname_dict[img_type]))
+                logger.info(chunk_fname_dict[img_type])
+
+                ia = casaStuff.image()
+                im = ia.imageconcat(outfile=fname_dict[img_type],
```

Comment:
I took a scan through and tried to follow but it seems like there's a risk of the end_chan from one chunk being the same as the start_chan for the next and being imaged  I don't see exactly where this is dropped or if there is an overlap just from the code alone.  Anyways, this whole PR doesn't seem like it would work if there's actually a problem at all so I think it's just a case of I'm not seeing where it's bookkept.

Reply:
That check gets handled in `CleanCall.return_chunked_channel_ranges` in this line: https://github.com/akleroy/phangs_imaging_scripts/pull/228/files#diff-b44a497b1b87cd2eae0ee857b74cf7453d91a88f47f54ac13f931c2649b59126R253

(This was actually an issue that kept cropping up when splitting the MS up, but I'm not getting any issues when running the chunks from the original whole MS).